### PR TITLE
Re-install census_filters

### PR DESCRIPTION
**Pull request by maxwell-demon**
The census_filter is mostly no-op for now.

### CODE REVIEW

File: build.json [hunk 1]
```diff
@@ -216,6 +216,7 @@
       ],
       "src": [
         "src/core/census/grpc_context.c",
+        "src/core/channel/census_filter.c",
```
Comment (a-veitch) [inline build.json]: ctiller@, what is the preferred location for filters like this - in the channel directory (current) or in package specific directories (census in this case). I think I'd prefer it in census, but not strongly.

File: src/core/channel/census_filter.c [hunk 2]
```diff
@@ -168,29 +154,33 @@ static void server_init_call_elem(grpc_call_element* elem,
...
+  grpc_iomgr_closure_init(d->on_done_recv, server_on_done_recv, elem);
   if (initial_op) server_mutate_op(elem, initial_op);
 }
 
 static void server_destroy_call_elem(grpc_call_element* elem) {
   call_data* d = elem->call_data;
+  census_stat stats[3];
   GPR_ASSERT(d != NULL);
-  d->stats.elapsed_time_ms = gpr_timespec_to_micros(
+  stats[0].id = CENSUS_RPC_SERVER_REQUESTS;
```
Comment (a-veitch) [inline src/core/channel/census_filter.c]: Can we make ths even more of a no-op - rmeove all the current census code, and just leave the filter framework?

File: build.json (hunk 1 above)
Comment (maxwell-demon) [inline build.json]: Good point, I see security filters are in core/security directory, maybe census_filters should also be in census directory. I'll let you decide, Craig.

File: src/core/channel/census_filter.c (hunk 2 above)
Comment (maxwell-demon) [inline src/core/channel/census_filter.c]: done

**Comment by maxwell-demon** [comment]
Per offline discussions, I have moved census_filters from channel to census directory.
PTAL.

**Comment by a-veitch** [comment]
What about naming the file census/grpc_filter.c instead of census/census_filter.c? The reasoning being that the 2xcensus in the name is redundant (current form) and it is a census-specific grpc filter (the former).

**Comment by maxwell-demon** [comment]
done. ptal.

**Comment by a-veitch** [comment]
Trying to figure out why the build is so badly broken - looks like a config error (not this PR) but want to be sure before merging. Is there anyway to retry? Maybe merge with current head first?

**Comment by maxwell-demon** [comment]
Looks like the build has been broken for a little while. I checked https://github.com/grpc/grpc/pulls, most of recent PRs have Jenkins failed.  

Clicking the "Details" link, I found that grpc-test instance on apphosting is very slow.  Checking the test status of different PRs, I found tests failed to build at various stages. e.g.

a) failed to check out from git:

> git checkout -f 0aef447888203e90feb87ffb107569ce4eb92e72
FATAL: Could not checkout 0aef447888203e90feb87ffb107569ce4eb92e72
    
b) failed to find  test shell file:

>/tmp/hudson3152823321678362373.sh: 2: /tmp/hudson3152823321678362373.sh: tools/jenkins/run_jenkins.sh: not found

I think there might be some problem with Jenkins instance rather than some code from HEAD broke the build.

**Comment by ctiller** [comment]
Looking into Jenkins problems.

**Comment by a-veitch** [comment]
Can I merge this now?

**Comment by maxwell-demon** [comment]
All c/c++ tests (I don't have node, csharp, and ruby installed on my workstation) can pass locally.  Craig, should we merge?

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* One or more commits are not associated with a GitHub user. The email used in the Git commit must match a verified email address in a GitHub user profile. For more information, please see https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/

_Regards,
CLA GitHub bot_

**Pull request merged**